src/main/java/com/mongodb/kafka/connect/util/config/BsonTimestampParser.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.util.config;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.slf4j.Logger;
+
+import org.bson.BsonTimestamp;
+import org.bson.json.JsonReader;
+
+import com.mongodb.lang.Nullable;
+
+public final class BsonTimestampParser {
+  public static final String FORMAT_DESCRIPTION =
+      "Must be either an integer number of seconds since the Epoch in the decimal format (example: 30),"
+          + " or an instant in the ISO-8601 format with one second precision (example: '1970-01-01T00:00:30Z'),"
+          + " or a BSON Timestamp in the canonical extended JSON (v2) format"
+          + " (example: '{\"$timestamp\": {\"t\": 30, \"i\": 0}}').";
+
+  public static BsonTimestamp parse(
```

Comment:
This class should have its own stand alone unit tests. Its not obvious that there are existing tests in `MongoSourceConfigTest` - they probably can just be moved over to `BsonTimestampParserTest`.

This way in the future maintainers will find it easier should any changes be required.

Reply:
This thread is a duplicate of https://github.com/mongodb/mongo-kafka/pull/125#discussion_r1016544358. Done.